### PR TITLE
Client Connection Manager

### PR DESCRIPTION
**Pull request by jacquese**
#### Description
Introduces a client connection manager which maintains the current client connections a server knows about. The system uses a subview to know when a worker system entity is removed, and uses that to clean up client connections after they have disconnected.

GDK-side heartbeating is removed as of this PR.

#### Tests
Gyms pass other than the possession gym which didn't pass on master either for me

#### Documentation
in-code docs

### CODE REVIEW

**Comment by jacquese** [comment]
https://improbableio.atlassian.net/browse/UNR-4861

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ClientConnectionManager.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+void ClientConnectionManager::RegisterClientConnection(const Worker_EntityId InWorkerEntityId, USpatialNetConnection* ClientConnection)
+{
+	WorkerConnections.Add(InWorkerEntityId, ClientConnection);
+}
+
+void ClientConnectionManager::CleanUpClientConnection(USpatialNetConnection* ConnectionCleanedUp)
+{
+	if (ConnectionCleanedUp->ConnectionClientWorkerSystemEntityId != SpatialConstants::INVALID_ENTITY_ID)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/ClientConnectionManager.cpp:47]: This can't be `INVALID_ENTITY_ID` anymore, can it?

**Review by improbable-valentyn — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Interop/ClientConnectionManager.cpp (hunk 1 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ClientConnectionManager.cpp:47]: hmm- not sure, will see

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ClientConnectionManager.cpp]: yeh i think you're right, removed

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/ClientConnectionManager.cpp]: turns out it totally can be invalid and removing the if here crashes the editor when a connection with an invalid entity ID makes its way in

**Pull request merged**